### PR TITLE
BUG: Fix bug with plot_alignment checks

### LINKED ISSUE #10308: misalignment of source space in forward sol
#### Describe the bug

Source space alignment in forward solution seems off.   

#### Steps to reproduce
code :
```python
# import parameters
from os.path import join
import mne
from mne import read_source_spaces, read_forward_solution

# get the data
root_dir = join('/', 'media', 'diptyajit', 'Ext_drive', 'MNE-sample-data', '')
subjects_dir = join(root_dir, 'subjects', '')

# read source space
src_fname = join(subjects_dir, 'sample', 'bem', 'sample-oct-6-src.fif')
src = read_source_spaces(src_fname)

# read forward sol
fwd_fname = join(root_dir, 'MEG', 'sample', 'sample_audvis-meg-eeg-oct-6-fwd.fif')
fwd = read_forward_solution(fwd_fname)

print('src output:', src)
print('forward_out:', fwd)

# 1st check source space alignment with original src
fig = mne.viz.plot_alignment(subject='sample', subjects_dir=subjects_dir,
                             surfaces='white', coord_frame='mri',
                             src=src)
mne.viz.set_3d_view(fig, azimuth=173.78, elevation=101.75, distance=0.30, focalpoint=(-0.03, -0.01, 0.03))

# 2nd check source space alignment with src in forward object
fig = mne.viz.plot_alignment(subject='sample', subjects_dir=subjects_dir,
                             surfaces='white', coord_frame='mri',fwd=fwd,
                             src=fwd['src'])
mne.viz.set_3d_view(fig, azimuth=173.78, elevation=101.75, distance=0.30, focalpoint=(-0.03, -0.01, 0.03))
```
#### Expected results
No difference in alignments 

#### Actual results
origina_src

forward_src

I know that number of source points can different in original source space vs in forward sol.  But why do I get a different alignment? Not sure if its a visualization issue or am I missing something.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #10308

@dasdiptyajit feel free to see if this fixes your problem

Our input validation for `trans` was deficient, this fixes it. cc @alexrockhill since I think this was your code at some point.

It also touches all tutorials that use `plot_alignment` to see if it breaks anything...

### CODE REVIEW

**Comment by alexrockhill** [comment]
Ah, I'm very sorry if I broke the forward solution plotting in my refactoring, it wasn't what I was focused on so I didn't check it as thoroughly. I'll review as soon as I can today.

File: mne/viz/_3d.py [hunk 1]
```diff
@@ -749,7 +755,7 @@ def plot_alignment(info=None, trans=None, subject=None, subjects_dir=None,
 
     if src is not None:
         atlas_ids, colors = read_freesurfer_lut()
-        for ss in src:
+        for si, ss in enumerate(src):
```
Comment (mmagnuski) [inline mne/viz/_3d.py:758]: The `si` variable does not seem to be used.

**Review by agramfort — APPROVED** [review]
LGTM

merge if happy @larsoner

File: examples/time_frequency/source_power_spectrum_opm.py [hunk 2]
```diff
@@ -8,7 +8,8 @@
 Here we compute the resting state from raw for data recorded using
 a Neuromag VectorView system and a custom OPM system.
 The pipeline is meant to mostly follow the Brainstorm :footcite:`TadelEtAl2011`
-`OMEGA resting tutorial pipeline <bst_omega_>`_.
+`OMEGA resting tutorial pipeline
+<https://neuroimage.usc.edu/brainstorm/Tutorials/RestingOmega>`__.
```
Comment (alexrockhill) [inline examples/time_frequency/source_power_spectrum_opm.py:12]: Too many underscores?

**Comment by alexrockhill** [comment]
Looks good, let me check all the plot alignment tutorial plots vs 0.24 though, one minute

**Comment by alexrockhill** [comment]
0.24 https://mne.tools/stable/auto_examples/preprocessing/movement_detection.html?highlight=plot_alignment
This PR https://41451-1301584-gh.circle-artifacts.com/0/dev/auto_examples/preprocessing/movement_detection.html?highlight=plot_alignment

File: examples/time_frequency/source_power_spectrum_opm.py (hunk 2 above)
Comment (larsoner) [reply examples/time_frequency/source_power_spectrum_opm.py:12]: That was intentional, it's an anonymous hyperlink in RST (which IIUC is for when you don't plan to reuse the same link name later)

**Comment by larsoner** [comment]
Do not compare to 0.24 but to `main`

https://mne.tools/dev/auto_examples/preprocessing/movement_detection.html?highlight=plot_alignment

Bug exists there, too, so not due to this PR

**Comment by alexrockhill** [comment]
This one looks off too PR: https://41451-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/3d_to_2d.html?highlight=plot_alignment

It's off in main too: https://mne.tools/dev/auto_examples/visualization/3d_to_2d.html?highlight=2d%203d
But not in 0.24

File: examples/visualization/3d_to_2d.py [hunk 3]
```diff
@@ -6,9 +6,8 @@
 ====================================================
```
Comment (alexrockhill) [inline examples/visualization/3d_to_2d.py:6]: If the period could be removed here, that would also be nice

**Comment by larsoner** [comment]
Let's fix those separately (#10312) and just fix things here that this PR breaks, if any

**Comment by alexrockhill** [comment]
Ok, good to merge by me, separate PRs sound good

**Pull request merged**